I test falliscono a causa del modulo `account_edi_ubl_cii` che viene installato per una catena di `auto_install` che parte da `accout_edi`.
Riporto l'errore per esteso:

```
2024-10-01 13:45:11,833 600 ERROR odoo odoo.addons.l10n_it_fatturapa_out.tests.test_fatturapa_xml_validation: ERROR: TestFatturaPAXMLValidation.test_invalid_char_fail
Traceback (most recent call last):
  File "/__w/l10n-italy/l10n-italy/l10n_it_fatturapa_out/tests/test_fatturapa_xml_validation.py", line 952, in test_invalid_char_fail
    invoice._post()
  File "/__w/l10n-italy/l10n-italy/l10n_it_ricevute_bancarie/models/account.py", line 150, in _post
    return super()._post(soft=soft)
  File "/__w/l10n-italy/l10n-italy/l10n_it_declaration_of_intent/models/account_move.py", line 81, in _post
    posted = super()._post(soft)
  File "/__w/l10n-italy/l10n-italy/l10n_it_account_stamp/models/account_move.py", line 154, in _post
    res = super(AccountMove, self)._post(soft=soft)
  File "/opt/odoo/addons/sale/models/account_invoice.py", line 75, in _post
    posted = super()._post(soft)
  File "/opt/odoo/addons/purchase_stock/models/account_invoice.py", line 173, in _post
    return super()._post(soft)
  File "/opt/odoo/addons/stock_account/models/account_move.py", line 53, in _post
    posted = super()._post(soft)
  File "/opt/odoo-venv/lib/python3.6/site-packages/odoo/addons/account_invoice_inter_company/models/account_move.py", line 72, in _post
    res = super()._post(soft=soft)
  File "/opt/odoo/addons/account_edi/models/account_move.py", line 361, in _post
    posted.edi_document_ids._process_documents_no_web_services()
  File "/opt/odoo/addons/account_edi/models/account_edi_document.py", line 249, in _process_documents_no_web_services
    self._process_jobs(self._convert_to_old_jobs_format(jobs))
  File "/opt/odoo/addons/account_edi/models/account_edi_document.py", line 145, in _process_jobs
    self._process_job(documents, doc_type)
  File "/opt/odoo/addons/account_edi/models/account_edi_document.py", line 231, in _process_job
    edi_result = edi_format._post_invoice_edi(documents.move_id, test_mode=test_mode)
  File "/opt/odoo/addons/account_edi_ubl_cii/models/account_edi_format.py", line 112, in _post_invoice_edi
    xml_content, errors = builder._export_invoice(invoice)
  File "/opt/odoo/addons/account_edi_ubl_cii/models/account_edi_xml_cii_facturx.py", line 242, in _export_invoice
    return etree.tostring(cleanup_xml_node(xml_content), xml_declaration=True, encoding='UTF-8'), set(errors)
  File "/opt/odoo/odoo/tools/xml_utils.py", line 107, in cleanup_xml_node
    xml_node = etree.fromstring(xml_node)
  File "src/lxml/lxml.etree.pyx", line 3213, in lxml.etree.fromstring (src/lxml/lxml.etree.c:79003)
  File "src/lxml/parser.pxi", line 1848, in lxml.etree._parseMemoryDocument (src/lxml/lxml.etree.c:118334)
  File "src/lxml/parser.pxi", line 1736, in lxml.etree._parseDoc (src/lxml/lxml.etree.c:117014)
  File "src/lxml/parser.pxi", line 1102, in lxml.etree._BaseParser._parseDoc (src/lxml/lxml.etree.c:111258)
  File "src/lxml/parser.pxi", line 595, in lxml.etree._ParserContext._handleParseResultDoc (src/lxml/lxml.etree.c:105102)
  File "src/lxml/parser.pxi", line 706, in lxml.etree._handleParseResult (src/lxml/lxml.etree.c:106810)
  File "src/lxml/parser.pxi", line 635, in lxml.etree._raiseParseError (src/lxml/lxml.etree.c:105664)
  File "<string>", line 40
lxml.etree.XMLSyntaxError: PCDATA invalid Char value 2, line 40, column 35
 ```

--- a/l10n_it_fatturapa_out/README.rst
+++ b/l10n_it_fatturapa_out/README.rst
@@ -114,7 +114,7 @@ Contributors
 * Lorenzo Battistini <https://github.com/eLBati>
 * Roberto Onnis
 * Alessio Gerace
-* Alex Comba
+* Alex Comba <alex.comba@agilebg.com>
 * Sergio Zanchetta <https://github.com/primes2h>
 * Marco Colombo <https://github.com/TheMule71>
 * Giovanni Serra <giovanni@gslab.it>

--- a/l10n_it_fatturapa_out/readme/CONTRIBUTORS.rst
+++ b/l10n_it_fatturapa_out/readme/CONTRIBUTORS.rst
@@ -2,7 +2,7 @@
 * Lorenzo Battistini <https://github.com/eLBati>
 * Roberto Onnis
 * Alessio Gerace
-* Alex Comba
+* Alex Comba <alex.comba@agilebg.com>
 * Sergio Zanchetta <https://github.com/primes2h>
 * Marco Colombo <https://github.com/TheMule71>
 * Giovanni Serra <giovanni@gslab.it>

--- a/l10n_it_fatturapa_out/static/description/index.html
+++ b/l10n_it_fatturapa_out/static/description/index.html
@@ -2,19 +2,18 @@
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
-<meta name="generator" content="Docutils: https://docutils.sourceforge.io/" />
+<meta name="generator" content="Docutils: http://docutils.sourceforge.net/" />
 <title>ITA - Fattura elettronica - Emissione</title>
 <style type="text/css">
 
 /*
 :Author: David Goodger (goodger@python.org)
-:Id: $Id: html4css1.css 9511 2024-01-13 09:50:07Z milde $
+:Id: $Id: html4css1.css 7952 2016-07-26 18:15:59Z milde $
 :Copyright: This stylesheet has been placed in the public domain.
 
 Default cascading style sheet for the HTML output of Docutils.
-Despite the name, some widely supported CSS2 features are used.
 
-See https://docutils.sourceforge.io/docs/howto/html-stylesheets.html for how to
+See http://docutils.sf.net/docs/howto/html-stylesheets.html for how to
 customize this style sheet.
 */
 
@@ -275,7 +274,7 @@ pre.literal-block, pre.doctest-block, pre.math, pre.code {
   margin-left: 2em ;
   margin-right: 2em }
 
-pre.code .ln { color: gray; } /* line numbers */
+pre.code .ln { color: grey; } /* line numbers */
 pre.code, code { background-color: #eeeeee }
 pre.code .comment, code .comment { color: #5C6576 }
 pre.code .keyword, code .keyword { color: #3B0D06; font-weight: bold }
@@ -301,7 +300,7 @@ span.option {
 span.pre {
   white-space: pre }
 
-span.problematic, pre.problematic {
+span.problematic {
   color: red }
 
 span.section-subtitle {
@@ -451,7 +450,7 @@ If you spotted it first, help us to smash it by providing a detailed and welcome
 <li>Lorenzo Battistini &lt;<a class="reference external" href="https://github.com/eLBati">https://github.com/eLBati</a>&gt;</li>
 <li>Roberto Onnis</li>
 <li>Alessio Gerace</li>
-<li>Alex Comba</li>
+<li>Alex Comba &lt;<a class="reference external" href="mailto:alex.comba&#64;agilebg.com">alex.comba&#64;agilebg.com</a>&gt;</li>
 <li>Sergio Zanchetta &lt;<a class="reference external" href="https://github.com/primes2h">https://github.com/primes2h</a>&gt;</li>
 <li>Marco Colombo &lt;<a class="reference external" href="https://github.com/TheMule71">https://github.com/TheMule71</a>&gt;</li>
 <li>Giovanni Serra &lt;<a class="reference external" href="mailto:giovanni&#64;gslab.it">giovanni&#64;gslab.it</a>&gt;</li>
@@ -468,9 +467,7 @@ If you spotted it first, help us to smash it by providing a detailed and welcome
 <div class="section" id="maintainers">
 <h2><a class="toc-backref" href="#toc-entry-7">Maintainers</a></h2>
 <p>This module is maintained by the OCA.</p>
-<a class="reference external image-reference" href="https://odoo-community.org">
-<img alt="Odoo Community Association" src="https://odoo-community.org/logo.png" />
-</a>
+<a class="reference external image-reference" href="https://odoo-community.org"><img alt="Odoo Community Association" src="https://odoo-community.org/logo.png" /></a>
 <p>OCA, or the Odoo Community Association, is a nonprofit organization whose
 mission is to support the collaborative development of Odoo features and
 promote its widespread use.</p>

--- a/l10n_it_fatturapa_out/tests/test_fatturapa_xml_validation.py
+++ b/l10n_it_fatturapa_out/tests/test_fatturapa_xml_validation.py
@@ -1,6 +1,6 @@
 # Copyright 2014 Davide Corio
 # Copyright 2015-2016 Lorenzo Battistini - Agile Business Group
-# Copyright 2018-2019 Alex Comba - Agile Business Group
+# Copyright 2018-2024 Alex Comba - Agile Business Group
 # Copyright 2024 Simone Rubino - Aion Tech
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
@@ -8,6 +8,7 @@ import base64
 import re
 from unittest.mock import Mock
 
+from lxml import etree
 from psycopg2 import IntegrityError
 
 import odoo
@@ -937,6 +938,39 @@ class TestFatturaPAXMLValidation(FatturaPACommon):
         finally:
             # Remove from the stack to not interfere with other tests
             odoo.http._request_stack.pop()
+
+    def test_invalid_char_fail(self):
+        """
+        - create an invoice with a product line with invalid char
+          as the https://en.wikipedia.org/wiki/List_of_Unicode_characters#Control_codes
+
+        expect to fail with a proper message
+        """
+        invoice = self._create_invoice()
+        # Set invoice line label to https://www.ascii-code.com/character/%E2%90%82
+        CHAR_CODE = 2
+        invoice.invoice_line_ids.name = chr(CHAR_CODE)
+        try:
+            invoice._post()
+        except etree.XMLSyntaxError as e:
+            entry = e.error_log.last_error
+            if entry.type_name == "ERR_INVALID_CHAR":
+                # just passing in order to not break the test execution
+                # due to the exception raised by account_edi_xml_cii_facturx.py/_export_invoice
+                # (https://github.com/odoo/odoo/blob/14.0/addons/account_edi_ubl_cii/
+                #  models/account_edi_xml_cii_facturx.py#L242)
+                pass
+        wizard = self.wizard_model.create({})
+        with self.assertRaises(UserError) as ue:
+            wizard.with_context({"active_ids": [invoice.id]}).exportFatturaPA()
+        error_msg = ue.exception.args[0]
+        error_fragments = (
+            f"Error processing invoice(s) {invoice.name}",
+            f"PCDATA invalid Char value {CHAR_CODE}",
+            "Line affected",
+        )
+        for fragment in error_fragments:
+            self.assertIn(fragment, error_msg)
 
     def test_multicompany_fail(self):
         """

--- a/l10n_it_fatturapa_out/wizard/efattura.py
+++ b/l10n_it_fatturapa_out/wizard/efattura.py
@@ -1,5 +1,6 @@
 # Copyright 2020 Giuseppe Borruso
 # Copyright 2020 Marco Colombo
+# Copyright 2024 Alex Comba - Agile Business Group
 import logging
 import os
 from datetime import datetime
@@ -254,10 +255,30 @@ class EFatturaOut:
         content = env.ref(
             "l10n_it_fatturapa_out.account_invoice_it_FatturaPA_export"
         )._render(template_values)
-        # 14.0 - occorre rimuovere gli spazi tra i tag
-        root = etree.fromstring(content, parser=etree.XMLParser(remove_blank_text=True))
-        # già che ci siamo, validiamo con l'XMLSchema dello SdI
-        errors = list(fpa_schema.iter_errors(root))
+        try:
+            # 14.0 - occorre rimuovere gli spazi tra i tag
+            root = etree.fromstring(
+                content, parser=etree.XMLParser(remove_blank_text=True)
+            )
+            # già che ci siamo, validiamo con l'XMLSchema dello SdI
+            errors = list(fpa_schema.iter_errors(root))
+        except etree.XMLSyntaxError as e:
+            entry = e.error_log.last_error
+            error_msg = _(
+                "Error processing invoice(s) %(invoices)s.\n\n"
+                "Error message: %(error)s\n\n",
+                invoices=", ".join(
+                    inv.display_name for inv in template_values["invoices"]
+                ),
+                error=entry.message,
+            )
+            # when there is a char that belongs to
+            # https://en.wikipedia.org/wiki/List_of_Unicode_characters#Control_codes
+            if entry.type_name == "ERR_INVALID_CHAR":
+                lines = [line for line in content.decode().split("\n")]
+                line = lines[entry.line - 1]
+                error_msg += _("Line affected:  %(line)s\n", line=line)
+            raise UserError(error_msg)
         if errors:
             # XXX - da migliorare?
             # i controlli precedenti dovrebbero escludere errori di sintassi XML

